### ISSUE TITLE
Create Arbitrary Child of Element

### ISSUE THREAD

**Issue by JKillian** [adobe-webplatform/Snap.svg#404]
Is there a way to create an arbitrary element of an Element? The only way to do this I could find [was this](https://groups.google.com/d/msg/snapsvg/D-8S1PWjA4k/g7hS4X-9FMYJ):

```
Snap.plugin(function (Snap, Element, Paper, global) {
      Element.prototype.el = Paper.prototype.el;
});
```

I want to create an `animateTransform` element inside of a `path` but can't figure out a good way to do it.

**Comment by ibrierley** [comment]
You could possibly do something like...

Paper.prototype.markupAnimateTransform = function( attr ) {
		return this.el('animateTransform').attr( <setMyAttributes> );
	};

Be aware, animating by animateTransform tag/element can be hit and miss, I think Firefox has some issues with it in some cases, so its likely not the way to go in many cases, where its best to use Snap animation instead. But there are a couple of cases where you want it in an svg image or something. Just make sure you are using it when you couldn't use Snap for the animation.

**Comment by JKillian** [comment]
Thanks for the advice @ibrierley. I actually could use Snap for the animation in all cases, I just figured using the `animateTransform` tag would be more performant (not Snap's fault on that).

**Comment by ibrierley** [comment]
No problem, just be aware of the compatibility issues (thinking about it, I'm not sure IE9/10 support it either, you may want to check, if it a requirement for you).

If performance is key, you may want to look at css (or combining with Velocity.js is straightforward, but it does mean another lib).

**Comment by JKillian** [comment]
Yeah, no version of IE supports it, but unfortunately they don't support CSS animations for SVG elements either. So when it comes to them, basically have to use  library like Snap :smiley:
